### ISSUE TITLE
[FEATURE] Web-sdk: Access to provider status inside the provider

### ISSUE THREAD

**Issue by thomaspoignant** [open-feature/js-sdk#893]
### Requirements

With the web-sdk `v1.0.0`, the provider status is now managed directly by the SDK which is great, but there are some usecases when it will be interesting to have access to the provider status inside the SDK.

One of the use cases I have is being able to send a `CACHED` reason when the provider is `STALE`.

**Comment by weyert** [comment]
Totally agree with this, probably worth doing a spec update for this?

**Comment by thomaspoignant** [comment]
> Totally agree with this, probably worth doing a spec update for this?

Not sure what the others think but it could be a great idea to spec it.

**Comment by thomaspoignant** [comment]
> You mean read-only access, I guess based on your use-case? I think I could be convinced of that. 🤔

Yes read-only, I don't need to change the status but I need to have access to it.

**Comment by beeme1mr** [comment]
It might be slightly complicated to get the status due to `domains`. Perhaps we should quickly prototype it to see how it could be implemented and then used in a provider.
